MapboxCoreNavigationTests.testJumpAheadToLastStep fails – RouterException Route Unknown Error

[This build](https://app.circleci.com/pipelines/github/mapbox/mapbox-navigation-ios/8088/workflows/2f875c15-1b80-4b7f-818e-1e7229c05cc0/jobs/69856) consistently fails with the following error:

```
2022-05-07 01:15:14.568030+0000 xctest[3222:81360] [Mapbox] [Info, nav-native]: RouteAlternativesControllerImpl::setRoutes alternatives=[ 1 2 ], removed=[ ]
2022-05-07 01:15:14.684274+0000 xctest[3222:81360] [Mapbox] [Info, nav-native]: Onboard router: Get route request: https://api.mapbox.com/directions/v5/mapbox/driving-traffic/-122.412755,37.792997%3B-122.433378,37.772701?alternatives=true&continue_straight=true&roundabout_exits=true&enable_refresh=true&avoid_maneuver_radius=623.211&geometries=polyline6&overview=full&steps=true&language=en-US&voice_instructions=true&voice_units=imperial&banner_instructions=true&bearings=170.859,90.0;&annotations=duration,maxspeed,congestion_numeric&access_token=****Bede
2022-05-07 01:15:14.693358+0000 xctest[3222:81360] [Mapbox] [Warning, valhalla]: No metadata found for tiles
2022-05-07 01:15:15.639997+0000 xctest[3222:81360] [Mapbox] [Error, nav-native]: getRoute("https://api.mapbox.com/directions/v5/mapbox/driving-traffic/-122.412755,37.792997%3B-122.433378,37.772701?alternatives=true&continue_straight=true&roundabout_exits=true&enable_refresh=true&avoid_maneuver_radius=623.211&geometries=polyline6&overview=full&steps=true&language=en-US&voice_instructions=true&voice_units=imperial&banner_instructions=true&bearings=170.859,90.0;&annotations=duration,maxspeed,congestion_numeric&access_token=pk.feedCafeDadeDeadBeef-BadeBede.FadeCafeDadeDeed-BadeBede") failed: RouterException Route Unknown Error
/Users/distiller/project/Tests/MapboxCoreNavigationTests/MapboxCoreNavigationTests.swift:209: error: -[MapboxCoreNavigationTests.MapboxCoreNavigationTests testJumpAheadToLastStep] : Asynchronous wait failed: Exceeded timeout of 5 seconds, with unfulfilled expectations: "Expect notification 'RouteControllerDidPassSpokenInstructionPoint' from <MapboxCoreNavigation.RouteController: 0x7fef856a0120>".
```

/cc @mapbox/navigation-ios

--- a/Tests/MapboxCoreNavigationTests/MapboxCoreNavigationTests.swift
+++ b/Tests/MapboxCoreNavigationTests/MapboxCoreNavigationTests.swift
@@ -85,13 +85,13 @@ class MapboxCoreNavigationTests: TestCase {
     }
     
     func testDepart() {
-        Navigator.shared.rerouteController.reroutesProactively = false
         navigation = MapboxNavigationService(routeResponse: response,
                                              routeIndex: 0,
                                              routeOptions: routeOptions,
                                              customRoutingProvider: MapboxRoutingProvider(.offline),
                                              credentials: Fixture.credentials,
                                              simulating: .never)
+        Navigator.shared.rerouteController.reroutesProactively = false
         
         // Coordinates from first step
         let coordinates = route.legs[0].steps[0].shape!.coordinates
@@ -124,7 +124,6 @@ class MapboxCoreNavigationTests: TestCase {
     }
     
     func testNewStep() {
-        Navigator.shared.rerouteController.reroutesProactively = false
         let steps = route.legs[0].steps
         // Create list of coordinates, which includes all coordinates in the first step and
         // first coordinate in the second step.
@@ -144,6 +143,7 @@ class MapboxCoreNavigationTests: TestCase {
                                                         credentials: Fixture.credentials,
                                                         locationSource: locationManager,
                                                         simulating: .never)
+        Navigator.shared.rerouteController.reroutesProactively = false
         
         var receivedSpokenInstructions: [String] = []
         
@@ -181,7 +181,6 @@ class MapboxCoreNavigationTests: TestCase {
     }
     
     func testJumpAheadToLastStep() {
-        Navigator.shared.rerouteController.reroutesProactively = false
         let coordinates = route.legs[0].steps.map { $0.shape!.coordinates }.flatMap { $0 }
         
         let now = Date()
@@ -202,6 +201,7 @@ class MapboxCoreNavigationTests: TestCase {
                                              credentials: Fixture.credentials,
                                              locationSource: locationManager,
                                              simulating: .never)
+        Navigator.shared.rerouteController.reroutesProactively = false
         
         expectation(forNotification: .routeControllerDidPassSpokenInstructionPoint,
                     object: navigation.router) { (notification) -> Bool in
@@ -334,7 +334,6 @@ class MapboxCoreNavigationTests: TestCase {
     }
     
     func testOrderOfExecution() {
-        Navigator.shared.rerouteController.reroutesProactively = false
         let trace = Fixture.generateTrace(for: route).shiftedToPresent().qualified()
         let locationManager = ReplayLocationManager(locations: trace)
         locationManager.speedMultiplier = 100
@@ -345,6 +344,7 @@ class MapboxCoreNavigationTests: TestCase {
                                              customRoutingProvider: MapboxRoutingProvider(.offline),
                                              credentials: Fixture.credentials,
                                              locationSource: locationManager)
+        Navigator.shared.rerouteController.reroutesProactively = false
         navigation.router.refreshesRoute = false
         
         struct InstructionPoint {

--- a/Tests/MapboxCoreNavigationTests/MapboxCoreNavigationTests.swift
+++ b/Tests/MapboxCoreNavigationTests/MapboxCoreNavigationTests.swift
@@ -33,6 +33,7 @@ class MapboxCoreNavigationTests: TestCase {
     }
     
     override func tearDown() {
+        Navigator.shared.rerouteController.reroutesProactively = true
         super.tearDown()
         navigation = nil
         UserDefaults.resetStandardUserDefaults()
@@ -84,6 +85,7 @@ class MapboxCoreNavigationTests: TestCase {
     }
     
     func testDepart() {
+        Navigator.shared.rerouteController.reroutesProactively = false
         navigation = MapboxNavigationService(routeResponse: response,
                                              routeIndex: 0,
                                              routeOptions: routeOptions,
@@ -122,6 +124,7 @@ class MapboxCoreNavigationTests: TestCase {
     }
     
     func testNewStep() {
+        Navigator.shared.rerouteController.reroutesProactively = false
         let steps = route.legs[0].steps
         // Create list of coordinates, which includes all coordinates in the first step and
         // first coordinate in the second step.
@@ -178,6 +181,7 @@ class MapboxCoreNavigationTests: TestCase {
     }
     
     func testJumpAheadToLastStep() {
+        Navigator.shared.rerouteController.reroutesProactively = false
         let coordinates = route.legs[0].steps.map { $0.shape!.coordinates }.flatMap { $0 }
         
         let now = Date()
@@ -330,6 +334,7 @@ class MapboxCoreNavigationTests: TestCase {
     }
     
     func testOrderOfExecution() {
+        Navigator.shared.rerouteController.reroutesProactively = false
         let trace = Fixture.generateTrace(for: route).shiftedToPresent().qualified()
         let locationManager = ReplayLocationManager(locations: trace)
         locationManager.speedMultiplier = 100

